presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -593,7 +594,27 @@ public synchronized void finishInsertIntoExistingPartition(ConnectorSession sess
     public synchronized Set<HivePrivilegeInfo> getTablePrivileges(String user, String databaseName, String tableName)
     {
         checkReadable();
-        return delegate.getTablePrivileges(user, databaseName, tableName);
+        SchemaTableName schemaTableName = new SchemaTableName(databaseName, tableName);
+        Action<TableAndMore> tableAction = tableActions.get(schemaTableName);
+        if (tableAction == null) {
+            return delegate.getTablePrivileges(user, databaseName, tableName);
+        }
+        switch (tableAction.getType()) {
+            case ADD:
+            case ALTER: {
+                if (!user.equals(tableAction.getData().getTable().getOwner())) {
+                    throw new PrestoException(NOT_SUPPORTED, "Cannot access a table newly created in the transaction with a different user");
+                }
+                List<PrivilegeGrantInfo> privilegeGrantInfoList = tableAction.getData().getPrincipalPrivilegeSet().getUserPrivileges().get(user);
```

Comment:
The owner privilege is calculated from the table owner field, so you will need something like this
```
        if (isTableOwner(user, databaseName, tableName)) {
            privileges.add(new HivePrivilegeInfo(OWNERSHIP, true));
        }